java/pattern/constraint/TypeConstraint.java

```diff
@@ -565,8 +566,17 @@ public boolean isPlays() {
 
         @Override
         public String toString() {
-            return PLAYS.toString() + SPACE + roleType +
-                    (overriddenRoleType != null ? "" + SPACE + AS + SPACE + overriddenRoleType : "");
+            String returnString = PLAYS.toString() + SPACE + roleType;
+            if (overriddenRoleType != null) {
+                String overriddenRoleTypeString;
+                if (overriddenRoleType.label().isPresent()) {
+                    overriddenRoleTypeString = overriddenRoleType.label().get().label();
+                } else {
+                    overriddenRoleTypeString = overriddenRoleType.reference().syntax();
+                }
```

Comment:
i wonder if we can compress this into a one liner with the `?` syntax and itll fit? probably not, but worth a try to see if it looks nicer :)